service/history/shardController.go

```diff
@@ -322,24 +322,31 @@ func (c *shardController) acquireShards() {
 	sw := c.metricsClient.StartTimer(metrics.HistoryShardControllerScope, metrics.AcquireShardsLatency)
 	defer sw.Stop()
 
-AcquireLoop:
-	for shardID := 0; shardID < c.config.NumberOfShards; shardID++ {
-		info, err := c.hServiceResolver.Lookup(string(shardID))
-		if err != nil {
-			c.logger.Error("Error looking up host for shardID", tag.Error(err), tag.OperationFailed, tag.ShardID(shardID))
-			continue AcquireLoop
-		}
+	concurrency := c.config.AcquireShardConcurrency()
```

Comment:
just for safety - enforce a lower bound of 1 i.e. concurrency = common.MaxInt(1, c.config.AcquireShards...())

Reply:
Done.